src/chart/pie/labelLayout.ts

```diff
@@ -228,6 +255,91 @@ function avoidOverlap(
     }
 }
 
+/**
+ * Set max width of each label, and then wrap each label to the max width.
+ *
+ * @param layout label layout
+ * @param availableWidth max width for the label to display
+ * @param forceRecalculate recaculate the text layout even if the current width
+ * is smaller than `availableWidth`. This is useful when the text was previously
+ * wrapped by calling `layoutText` but now `availableWidth` changed, in which
+ * case, previous wrapping should be redo. `forceRecalculate` is ignored if
+ * `labelStyleWidth` is set.
+ */
+function layoutText(
+    layout: LabelLayout,
+    availableWidth: number,
+    forceRecalculate: boolean = false
+) {
+    const label = layout.label;
+    const style = label.style;
+    const textRect = layout.rect;
+    const bgColor = style.backgroundColor;
+    const padding = style.padding as number[];
+    const paddingH = padding ? padding[1] + padding[3] : 0;
+    const overflow = style.overflow;
+
+    // textRect.width already contains paddingH if bgColor is set
+    const oldOuterWidth = textRect.width + (bgColor ? 0 : paddingH);
+    if (availableWidth < oldOuterWidth
+        || forceRecalculate && layout.labelStyleWidth == null
+    ) {
+        const oldHeight = textRect.height;
+        if (overflow && overflow.match('break')) {
+            const oldEllipsis = style.ellipsis;
+
+            // Temporarily set background to be null to calculate
+            // the bounding box without backgroud.
+            label.setStyle('backgroundColor', null);
+            label.setStyle('ellipsis', '');
+
+            // Set constraining width
+            label.setStyle('width', availableWidth - paddingH);
+
+            if (layout.labelStyleWidth == null) {
+                // This is the real bounding box of the text without padding
+                const innerRect = label.getBoundingRect();
+                innerRect.applyTransform(label.getComputedTransform());
```

Comment:
There is no need to applyTransform because only scale transform will affect the width, which doesn't exits in pie labels